Common/Helpers/BlobStorageClient.cs

```diff
@@ -0,0 +1,211 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Helpers
+{
+    /// <summary>
+    /// Helper methods, related to blob storage.
+    /// </summary>
+    public class BlobStorageClient : IBlobStorageClient
+    {
+        private CloudStorageAccount _storageAccount;
+        private CloudBlobContainer _container;
+        private CloudBlockBlob _blob;
+        private readonly CloudBlobClient _blobClient;
+        private readonly string _containerName;
+        private readonly string _blobName;
+
+        public BlobStorageClient(string connectionString, string containerName, string blobName)
+        {
+            _storageAccount = CloudStorageAccount.Parse(connectionString);
+            _blobClient = _storageAccount.CreateCloudBlobClient();
+            _containerName = containerName;
+            _blobName = blobName;
+        }
+        
+        public async Task UploadFromByteArrayAsync(byte[] buffer, int index, int count, AccessCondition accessCondition, BlobRequestOptions options, OperationContext operationContext)
+        {
+            await this.CreateCloudBlockBlobAsync();
+            await _blob.UploadFromByteArrayAsync(
+                buffer,
+                index,
+                count,
+                accessCondition,
+                options,
+                operationContext);
+        }
+        
+        public async Task<byte[]> GetBlobData()
+        {
+            await this.CreateCloudBlockBlobAsync();
+            bool exists = await _blob.ExistsAsync();
+            if (exists)
+            {
+                await _blob.FetchAttributesAsync();
+                long blobLength = _blob.Properties.Length;
+
+                if (blobLength > 0)
+                {
+                    byte[] existingBytes = new byte[blobLength];
+                    await _blob.DownloadToByteArrayAsync(existingBytes, 0);
+                    return existingBytes;
+                }
+            }
+            return null;
+        }
+
+        public async Task<string> GetBlobEtag()
+        {
+            await this.CreateCloudBlockBlobAsync();
```

Comment:
I think this is a little confusing. For someone that calls GetBlobEtag method they assume that the blob is already created and they want to access its etag. Same comment but I think this should be a 1 time operation: CreateCloudBlockBlobAsync